pipeline/deploy.sh

```diff
@@ -4,45 +4,116 @@ set +x
 
 if [[ -z "${VSPHERE_CSI_CONTROLLER_IMAGE}" ]]
 then
-	echo "Env variable unset: VSPHERE_CSI_CONTROLLER_IMAGE"
+	echo "Env variable unset: VSPHERE_CSI_CONTROLLER_IMAGE."
 	exit 1
 fi
 
 if [[ -z "${VSPHERE_SYNCER_IMAGE}" ]]
 then
-	echo "Env variable unset: VSPHERE_SYNCER_IMAGE"
+	echo "Env variable unset: VSPHERE_SYNCER_IMAGE."
 	exit 1
 fi
 
 # Borrow a testbed from CSI Testbed Pool Svc.
-if ! testbed=$(curl -X 'GET' "${CNS_TESTBEDPOOL_SVC_URL}");
+if ! testbed=$(curl -X 'PUT' "${CNS_TESTBEDPOOL_SVC_URL}" -H 'accept: application/json' -u "${CNS_MANAGER_USERNAME}:${CNS_MANAGER_PASSWORD}");
 then
-	echo "Unable to borrow a testbed!"
+	echo "Unable to borrow a testbed."
 	exit 1
 fi
-echo "TestbedInfo: $testbed"
 
+echo "TestbedInfo: $testbed ."
+
+# Extract testbed ID from borrow testbed API Response.
+if ! id=$(echo "$testbed" | jq '.id'|tr -d '"');
+then
+	echo "Error getting the id."
+	exit 1
+fi
+
+curl --request PUT --header "PRIVATE-TOKEN: $CNS_CSI_GIT_TOKEN" "https://gitlab.eng.vmware.com/api/v4/projects/$CNS_CSI_GIT_PROJECT_ID/variables/CNS_CSI_ID" --form "value=$id"
```

Comment:
Are we trying to set a variable here? If yes, wouldn't this complicate things when we run multiple CI pipelines concurrently?

Reply:
This method of storing the testbed ID is removed and adopted a new logic to store the testbed ID.

Introduced a variable field into .gitlab-ci.yml

  variables:
    CNS_CSI_ID: "${CNS_CSI_ID}"

Reply:
Changed the logic to store the testbed's ID in a file on the workspace.